File: backend/src/main/java/com/telemedicina/app/model/Doctor.java
package com.telemedicina.app.model;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import java.util.HashSet;
import java.util.Set;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Builder
public class Doctor{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne(orphanRemoval = true, cascade = CascadeType.ALL)
    private Persona persona;
    private String licencia;
    private double honorarios;
    @ManyToOne(cascade = CascadeType.PERSIST)
    private Especialidad especialidad;
    @ManyToMany(cascade = CascadeType.ALL)
    private Set<Paciente> pacientes = new HashSet<>();
}

File: frontend/src/app/core/services/api-client/models/doctor.ts
/* tslint:disable */
/* eslint-disable */
import { Especialidad } from '../models/especialidad';
import { Paciente } from '../models/paciente';
import { Persona } from '../models/persona';
export interface Doctor {
  especialidad?: Especialidad;
  honorarios?: number;
  id?: number;
  licencia?: string;
  pacientes?: Array<Paciente>;
  persona?: Persona;
}
